### PR TITLE
chore(server): change generate address

### PR DESCRIPTION
**Pull request by shaojunda**
1. check if the default lock script is used.
2. adjust tests.

### CODE REVIEW

File: server/test/models/address_test.rb [hunk 1]
```diff
@@ -34,13 +34,66 @@ class AddressTest < ActiveSupport::TestCase
     end
   end
 
-  test ".find_or_create_address should return the address when the address_hash exists" do
+  test "address_hash should be nil when args is invalid" do
+    VCR.use_cassette("blocks/10") do
```
Comment (duanyytop) [inline server/test/models/address_test.rb]: I want to know whether setting "blocks/10" as constant to reuse is a better action in ruby ?

File: server/config/initializers/ckb/utils.rb [hunk 2]
```diff
@@ -35,10 +35,9 @@ def self.miner_hash(cellbase)
     end
```
Comment (classicalliu) [inline server/config/initializers/ckb/utils.rb]: Move this file to `app` dir.

File: server/config/initializers/ckb/utils.rb [hunk 3]
```diff
@@ -48,6 +47,15 @@ def self.generate_address(lock_script)
       CKB::ConvertAddress.encode(Address::PREFIX_TESTNET, payload)
     end
 
+    def self.use_default_lock_script?(lock_script)
+      first_arg = lock_script.stringify_keys["args"].first
+      binary_hash = lock_script.stringify_keys["binary_hash"]
+
+      return false if first_arg.blank? || binary_hash.blank?
+
+      CKB::Utils.valid_hex_string?(first_arg) && binary_hash == LockScript::SYSTEM_SCRIPT_CELL_HASH
```
Comment (classicalliu) [inline server/config/initializers/ckb/utils.rb]: I think check `binary_hash` is enough

File: server/test/models/ckb_sync/persist_test.rb [hunk 4]
```diff
@@ -374,6 +374,43 @@ class PersistTest < ActiveSupport::TestCase
...
+        assert_nil local_block.miner_hash
+      end
+    end
+
+    test ".save_block generated block's miner hash should be nil when binary hash is invalid" do
+      VCR.use_cassette("blocks/10") do
+        SyncInfo.local_inauthentic_tip_block_number
+        node_block = CkbSync::Api.instance.get_block(DEFAULT_NODE_BLOCK_HASH).deep_stringify_keys
+        node_block["commit_transactions"].first["outputs"].first["lock"]["args"] = ["0xc30257c81dde7766fc98882ff1e9f8e95abbe79345982e12c6a849de90cbbad1"]
+        node_block["commit_transactions"].first["outputs"].first["lock"]["binary_hash"] = "0xc30257c81dde7766fc98882ff1e9f8e95abbe79345982e12c6a849de90cbbad1"
```
Comment (classicalliu) [inline server/test/models/ckb_sync/persist_test.rb]: Extract `node_block["commit_transactions"].first["outputs"].first["lock"]` maybe better.

File: server/test/models/address_test.rb (hunk 1 above)
Comment (shaojunda) [reply server/test/models/address_test.rb]: blocks/10 is point to a real file named `10.yml`, in test I think can be left as it is.

Comment (duanyytop) [reply server/test/models/address_test.rb]: OK

File: server/config/initializers/ckb/utils.rb (hunk 2 above)
Comment (shaojunda) [reply server/config/initializers/ckb/utils.rb]: ok I move it to lib.

File: server/config/initializers/ckb/utils.rb (hunk 3 above)
Comment (shaojunda) [reply server/config/initializers/ckb/utils.rb]: ok, Thank you.

File: server/test/models/ckb_sync/persist_test.rb (hunk 4 above)
Comment (shaojunda) [reply server/test/models/ckb_sync/persist_test.rb]: ok, I will fix this.

**Pull request merged**